### ISSUE TITLE
[FEATURE REQUEST] warning on using short arguments

### ISSUE THREAD

**Issue by pantocrator27** [koalaman/shellcheck#919]
#### Feature Request
This may be a little nebulous however recently I have been preferring long arguments to short arguments where applicable to increase readability in scripts.  I know that not everything supports long arguments which is why I think this should be a warning vs an error.  

#### For new checks and feature suggestions
- [X] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [X] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

**Comment by brother** [comment]
If we encounter a command followed by a short flag it should emit a warning about using a long flag instead? I do like the longer flag names but I have a hard time seeing how it would be implemented if it wasn't a list of if command X uses flag Y then warn and tellthem to use flag longY.
Warn on every short flag and not being sure there are long flags seems not very helpful.

```shell
cd -P my/dir/structure
      ^------- WARN: You are using a short flag here, please use the long version instead.
```

Then looking in the man page you'll notice that there are only short flags and have to silence the warning with a shellcheck directive.

```shell
# shellcheck disable=SCsomething242
cd -P my/dir/structure
```

This is borderline code style and religion. I am on your camp for the sentiment but I do think I would disable the check for my daily use.

**Comment by pantocrator27** [comment]
I agree that this is probably more religion ....
